apps/native-component-list/webpack.common.js

```diff
@@ -197,34 +240,21 @@ function useWebModule(modulePathToHiJack, redirectPath, initialRoot = 'expo/buil
 const publicPath = '/';
 
 module.exports = {
-  mode: 'development',
+  mode: environment,
+  context: __dirname,
   devtool: 'cheap-module-source-map',
-
-  entry: [require.resolve('react-dev-utils/webpackHotDevClient'), locations.appMain],
   // configures where the build ends up
   output: {
-    path: locations.output,
-    pathinfo: true,
+    path: locations.absolute(productionPath),
     filename: 'bundle.js',
```

Comment:
To turn on caching, you can content-address the filenames with:
```
filename: '[chunkhash].js',
chunkFilename: '[chunkhash.js]`,
hashFunction: 'sha512',
```

And then need to get the entrypoint hash/filename from the build stats and pass it into the HTML generator so that index.html can load the correct file. But this way we can turn on caching for JS files and get those performance numbers up.